### PR TITLE
Add client factory

### PR DESCRIPTION
**Pull request by jjok**
How about some kind of factory to make it easier to create a new MQTT client instance?

### CODE REVIEW

File: src/ClientFactory.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace oliverlorenz\reactphpmqtt;
+
+use oliverlorenz\reactphpmqtt\protocol\Version;
+use React\Dns\Resolver\Factory as DnsResolverFactory;
+use React\EventLoop\Factory as EventLoopFactory;
+
+class ClientFactory
+{
+    public static function createClient(Version $version, $resolverIp)
```
Comment (oliverlorenz) [inline src/ClientFactory.php]: can you give $resolverIp a default?

BTW: This resolver thing is something I'm not happy with, because its a unusual thing you have to set your DNS manually. Maybe you have a idea to change that? But, if we do that lets do this in another PR

File: README.md [hunk 2]
```diff
@@ -1,67 +1,39 @@
 # PHP MQTT Client
 
-phpMqttClient is a mqtt client library for PHP. Its based on the reactPHP socket-client and added the mqtt protocol specific functions. I hope its a better starting point that the existing php mqtt libraries. 
+phpMqttClient is an MQTT client library for PHP. Its based on the reactPHP
+socket-client and added the mqtt protocol specific functions. I hope its a
```
Comment (oliverlorenz) [inline README.md]: Why this lines are separated like this? :confused:

File: README.md [hunk 3]
```diff
@@ -1,67 +1,39 @@
...
-$connector = new oliverlorenz\reactphpmqtt\Connector($loop, $resolver, $version);
+$connector = ClientFactory::createClient(new Version4(), '8.8.8.8');
 
 $p = $connector->create($config['server'], $config['port'], $config['options']);
-$p->then(function(\React\Stream\Stream $stream) use ($connector) {
+$p->then(function(Stream $stream) use ($connector) {
     return $connector->publish($stream, 'a/b', 'example message');
 });
-$loop->run();
+$connector->getLoop()->run();
```
Comment (oliverlorenz) [inline README.md]: I Like that! And as I see this is not a breaking change. Only a better way to to the same in less lines :+1:

File: README.md (hunk 2 above)
Comment (jjok) [inline README.md]: I just split it on to two lines as it was running off the edge of my screen. I could undo the change if you prefer.

File: src/ClientFactory.php (hunk 1 above)
Comment (jjok) [inline src/ClientFactory.php]: What should the default be? `8.8.8.8`?

Actually, I don't really know why you have to select a DNS resolver manually. I'm not too familiar with how React works, but I can look into it.

**Comment by oliverlorenz** [comment]
To be honest: Its a long time ago, that I created the basements for the project. I can remember that this was the suggested way in the example s, but I don't know this is still state of the art.

I could imagine this is a reason why people are confused about the interface and how it's to use. I think to make this dependency optional would be a big step forward.

File: src/ClientFactory.php (hunk 1 above)
Comment (oliverlorenz) [inline src/ClientFactory.php]: >  I don't really know why you have to select a DNS resolver manually

Because it was the suggested way from the examples. But I would be very happy if this would be a optional dependency

Comment (jjok) [inline src/ClientFactory.php]: I'll just set the default to `8.8.8.8` for now. Then we can see if we can change how it works later.

I would've thought that you'd be able to just use whatever DNS resolver your web browser would usually use, but I don't really know how that works.

Comment (oliverlorenz) [inline src/ClientFactory.php]: I put that also to the roadmap. Like that as the first iteration improvement :)
Thank you for that :)
